metaserver/src/main/scala/scala/meta/languageserver/providers/SquiggliesProvider.scala

```diff
@@ -6,31 +6,35 @@ import com.typesafe.scalalogging.LazyLogging
 import scala.{meta => m}
 import langserver.messages.PublishDiagnostics
 import scala.meta.languageserver.ScalametaEnrichments._
+import scala.meta.languageserver.MonixEnrichments._
 import scala.tools.nsc.interpreter.OutputStream
 import monix.eval.Task
+import monix.execution.Scheduler
+import monix.reactive.Observable
 import org.langmeta.AbsolutePath
 
 class SquiggliesProvider(
-    configuration: Task[Configuration],
+    configuration: Observable[Configuration],
     cwd: AbsolutePath,
     stdout: OutputStream
-) extends LazyLogging {
+)(implicit s: Scheduler)
+    extends LazyLogging {
+  private val isEnabled: () => Boolean =
+    configuration.map(_.scalafix.enabled).toFunction0()
 
   lazy val linter = new Linter(cwd, stdout)
 
   def squigglies(doc: m.Document): Task[Seq[PublishDiagnostics]] =
     squigglies(m.Database(doc :: Nil))
-  def squigglies(db: m.Database): Task[Seq[PublishDiagnostics]] =
-    for {
-      config <- configuration
-    } yield {
+  def squigglies(db: m.Database): Task[Seq[PublishDiagnostics]] = Task.eval {
+    if (!isEnabled()) Nil
```

Comment:
`isEnabled` is based on `scalafix.enabled`. This check also skips the compiler errors, which are not related.